pkg/authenticator/authenticator.go

```diff
@@ -263,29 +264,29 @@ func (auth *Authenticator) ParseAuthenticationResponse(response []byte) error {
 	return nil
 }
 
-// waitForInjectedCertificateFile waits for a given timeout to see if the cert file
-// exists in the given path. If it's not there by the end of the timeout, it returns
+// waitForInjectedCertificateFile waits for retryCountLimit seconds to see if the cert file
+// exists in the given path. If it's not there by the end of the retry count limit, it returns
 // an error.
-func waitForInjectedCertificateFile(certificatePath string, timeout time.Duration) error {
-	deadline := time.Now().UTC().Add(timeout)
-	for {
+func waitForInjectedCertificateFile(certificatePath string, retryCountLimit int) error {
+	constantBackOff := backoff.NewConstantBackOff(time.Second)
+	count := 0
+	return backoff.Retry(func() error {
 		info, err := os.Stat(certificatePath)
-
 		if !os.IsNotExist(err) && !info.IsDir() {
-
 			// No error, the certificate exists within the deadline
 			return nil
 		}
 
-		if time.Now().After(deadline) {
-			// Timed out waiting for the certificate to exist
-			return log.RecordedError(log.CAKC033E, certificatePath)
+		count++
+		if count >= retryCountLimit {
+			return backoff.Permanent(log.RecordedError(log.CAKC033E, retryCountLimit, certificatePath))
```

Comment:
I'm not super familiar with this backoff library, but I do see it has a facility for maximum retry attempts: https://godoc.org/github.com/cenkalti/backoff#WithMaxRetries

Did you look at this and determine it didn't do what you want? Or is this something to consider rather than keeping our own count?

Reply:
i actually moved to another backoff that @abrahamko implemented. it will be in `utils`.